### DISCUSSION TITLE
Welcome to music_down Discussions!

### DISCUSSION THREAD

**Discussion by TotoWang-hhh** [TotoWang-hhh/music_down#5] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by zem128764-oss** [comment]
大佬我想学Python但是我不懂Python语言我要去哪里可以免费学？😭

**Reply by TotoWang-hhh** [reply]
比较长，因为我的答复很多时候就是这样，希望你能够看完。我可能会无意识地假设你是一位初学Python的学生，如果这不符合事实，你可以追问我，但请谅解我以自己的个人经历进行假设。

我自己是先在一个机构学了一些基础，之后一直都是自己尝试搓项目，边学边查，慢慢就都会了。

很抱歉我目前不能向你推荐任何编程培训机构，因为我曾经报课学习的那一家已经倒闭跑路了（这事情当时还闹得不小来着）。或许网路上可以免费观看的Python教程，以及一部分难度对口的Python实战训练，可以作为平替。总之，你会需要任何一个人或组织以任意形式直接教会你Python的基础语法。

在学会基本的编程知识和Python语法之后，可以按照自己的想法尝试做一些项目。不要急着做太过复杂的东西，也不要急着公开发布出来（发在GitHub是可以的，但现如今发在社交媒体或视频网站的话只会引来一堆喷子并打击你的自信心，我不太理解为什么当年我照抄网上的代码编写这个音乐下载器的初代版本时，就没有遇到过这些破事），遇到不会写的就去查。每一项技能，一开始都是复制粘贴，慢慢地变成照着网上的例子自己改，再慢慢地就学会了。我自己在初一初二的时候因为空闲时间比较多，加上教室离学校图书馆很近，每次一有空都会去图书馆，不为看书，就是为了占着公共电脑写一些看起来没啥用的小玩意，写着写着就啥都会了。

另外，我要强调的是，如果按照我的学习方式，几乎任何东西都是从照抄到照着别人的自己改，再到自己学会的，但是一定要自己去学，而不是永远地抄下去。

关于直到2023年前后才开始普及开来的AI：正如我说，目前广泛使用的大模型是2023年左右才普及开的，所以我学习编程的时代可谓是相当地清净。我个人是不太赞成使用AI进行大量编码的，使用AI应当只在两种场景：1. 用于查询，查到之后就全靠自己动手 2. 用于进行重复性或没有任何技术含量的编码工作。永远记住AI只是一个更好的搜索引擎或者说是卖苦力的，它不会是一个合格的脑力劳动者。

当然，如果你需要任何帮助或者有任何问题的话，都可以通过我的邮箱（在我的GitHub主页可查）或者其他方式（在我的个人网站可查）联系我，你也可以在这个issue留下联系方式（如果你觉得有必要的话）以便我找到你。

**Reply by zem128764-oss** [reply]
嘿嘿不学了好麻烦我发现让ai帮我写也行谢谢你老baby
